lib/ex_plasma/transaction.ex

```diff
@@ -111,14 +111,17 @@ defmodule ExPlasma.Transaction do
     do: new([[], tx_type, inputs, outputs, tx_data, metadata])
 
   def new([sigs, tx_type, inputs, outputs, tx_data, metadata]) do
-    %__MODULE__{
-      tx_type: tx_type,
-      sigs: sigs,
-      inputs: Enum.map(inputs, &Utxo.new/1),
-      outputs: Enum.map(outputs, &Utxo.new/1),
-      tx_data: tx_data,
-      metadata: metadata
-    }
+    with {:ok, inputs} <- build_utxos(inputs),
```

Comment:
all your tests are using one input or one output, write more complicated tests so that you thoroughly test `build_utxos`
there's also no negative tests cases for the error tuple in `build_utxos`.

Reply:
I think these would cover the error tuples? https://github.com/omisego/ex_plasma/pull/23/files#diff-c201e5caa4e06e66ab81c30bc7b318dcR7.

Reply:
Added: https://github.com/omisego/ex_plasma/pull/23/commits/49ce8d0d260472b1aa09831aba98b2f0f5431a4a